internal/handlers/common/filter.go

```diff
@@ -594,76 +594,231 @@ func filterFieldExprSize(fieldValue any, sizeValue any) (bool, error) {
 
 // filterFieldExprBitsAllClear handles {field: {$bitsAllClear: value}} filter.
 func filterFieldExprBitsAllClear(fieldValue, maskValue any) (bool, error) {
-	fieldBinary, maskBinary, err := getBinaryParams(fieldValue, maskValue)
-	if err != nil {
-		return false, formatBitwiseOperatorErr(err, "$bitsAllClear", maskValue)
-	}
+	switch value := fieldValue.(type) {
+	case int32:
+		bitmask, err := getBinaryMaskParam(maskValue)
+		if err != nil {
+			return false, formatBitwiseOperatorErr(err, "$bitsAllClear", maskValue)
+		}
+		return (^uint64(value) & bitmask) == bitmask, nil
+
+	case int64:
+		bitmask, err := getBinaryMaskParam(maskValue)
+		if err != nil {
+			return false, formatBitwiseOperatorErr(err, "$bitsAllClear", maskValue)
+		}
+		return (^uint64(value) & bitmask) == bitmask, nil
 
-	for i := 0; i < len(fieldBinary.B); i++ {
-		if (fieldBinary.B[i] & maskBinary.B[i]) != 0 {
+	case float64:
+		// TODO check float negative zero
+		if value != math.Trunc(value) || math.IsNaN(value) || math.IsInf(value, 0) || value > math.MaxInt64 {
```

Comment:
```suggestion
		if value != math.Trunc(value) || 
		   math.IsNaN(value) || 
		   math.IsInf(value, 0) || 
		   value > math.MaxInt64 {
```

Reply:
When editing if condition: adding or deleting, it will clearly and explicitly visible what was added, deleted or what condition was edited. 

Having single row for all condition it would be more difficult to catch the change in a row.